### PR TITLE
PHP 8: Code Review `SurveyQuestionPool`

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Hi @alex40724,

I completed the review of the `Modules/SurveyQuestionPool`.
This component looked very good, but of course I found some issues.

Results:
- [ ] The code style is `PSR-2 + X` compliant: 6 files have been changed by the `PHP-CS-FIXER`
- [ ] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`: If found 2 usages of `$_SESSION`
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [x] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties)

I also fixed some trivial issues reported by my inspection profile.

--

Actions needed:
- [ ] Please eliminate the usage of `$_SESSION`
- [ ] Please check my inline comments. Some of them are just information, some require an action. You can use `TODO PHP8-REVIEW` when searching.

Best regards,
@mjansenDatabay

### CODE REVIEW

File: Modules/SurveyQuestionPool/Questions/class.SurveyQuestionGUI.php [hunk 1]
```diff
@@ -38,8 +38,7 @@ abstract class SurveyQuestionGUI
     protected array $cumulated = [];
     protected string $parent_url = "";
     protected ilLogger $log;
-    
-    public $object;
+    public ?SurveyQuestion $object = null;
```
Comment (mjansenDatabay) [inline Modules/SurveyQuestionPool/Questions/class.SurveyQuestionGUI.php:41]: Not 100% sure here

File: Modules/SurveyQuestionPool/Phrases/class.ilSurveyPhrasesGUI.php [hunk 2]
```diff
@@ -82,7 +82,7 @@ public function executeCommand() : string
      */
     public function deletePhrase() : void
     {
-        $this->tpl->setOnScreenMessage('info');
+        $this->tpl->setOnScreenMessage('info');// TODO PHP8-REVIEW: Missing arguments here
```
Comment (mjansenDatabay) [inline Modules/SurveyQuestionPool/Phrases/class.ilSurveyPhrasesGUI.php:85]: Missing arguments here

File: Modules/SurveyQuestionPool/Editing/class.EditingGUIRequest.php [hunk 3]
```diff
@@ -25,7 +25,6 @@ class EditingGUIRequest
 {
     use BaseGUIRequest;
 
-    protected \ILIAS\HTTP\Services $http;
```
Comment (mjansenDatabay) [inline Modules/SurveyQuestionPool/Editing/class.EditingGUIRequest.php:28]: Already provided by the trait

**Comment by alex40724** [comment]
@mjansenDatabay Thanks, I took care of the empty info and the _session use.

**Pull request merged**